crates/revm/src/journaled_state.rs

```diff
@@ -544,9 +545,10 @@ impl JournaledState {
             ),
         };
         // preload storages.
-        for slot in slots {
-            if let Entry::Vacant(entry) = account.storage.entry(*slot) {
-                let storage = db.storage(address, *slot).map_err(EVMError::Database)?;
+        for storage_key in storage_keys {
+            let slot = U256::from_be_bytes(storage_key.0);
+            if let Entry::Vacant(entry) = account.storage.entry(slot) {
+                let storage = db.storage(address, slot).map_err(EVMError::Database)?;
```

Comment:
Instead of converting to a `U256` should we do a backwards incompatible change that makes the account storage a `HashMap<B256, U256>`?

This would match the expected behaviour of the Ethereum protocol.

Reply:
This does not make sense